code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -117,27 +118,19 @@
 #define REGALRAT_INTERACTION "regalrat"
 
 /mob/living/simple_animal/hostile/regalrat/AttackingTarget()
-	if (DOING_INTERACTION(src, REGALRAT_INTERACTION))
-		return
-	if (QDELETED(target))
+	if (DOING_INTERACTION(src, REGALRAT_INTERACTION) || QDELETED(target))
 		return
 	if(istype(target, /obj/machinery/door/airlock) && !opening_airlock)
 		pry_door(target)
 		return
-
-	if (target.reagents && target.is_injectable(src, allowmobs = TRUE) && !istype(target, /obj/item/food/cheese))
+	if (target.reagents && target.is_injectable(src, allowmobs = TRUE) && !istype(target, /obj/item/food/cheese) && !src.combat_mode && src.mind)
```

Comment:
this short-circuits, so re-arrange the order so that we can get the simplest (or those most likely to be falsey) checks out of the way the quickest. something like:

```suggestion
	if (src.mind && !src.combat_mode && target.reagents && target.is_injectable(src, allowmobs = TRUE) && !istype(target, /obj/item/food/cheese))
```

Reply:
That's a good call, tbh, I'll keep it in mind for future commits as well.